Add v0.29.0 upgrade guide (#1805)
pomerium/documentation: content/docs/deploy/upgrading.mdx

:::info

> Your Pomerium Enterprise version should always match the same **minor version number** as your Pomerium Core version. For example:

Reviewer: Maybe we can switch to explicit `<br>` tags instead?

See [OpenTelemetry tracing](/docs/reference/tracing) for configuration details and examples.

##### Removed Tracing Options

Existing OpenCensus tracing configurations are automatically removed upon upgrading Pomerium Console to version 0.29.0 or later.

#### Enterprise API Type Changes

Code generation of gRPC clients was simplified by removing the dependency on Envoy. This included several changes to the Enterprise API types in v0.29.0. If you are using Pomerium's [enterprise-client-go](https://github.com/pomerium/enterprise-client-go) or [enterprise-client-python](https://github.com/pomerium/enterprise-client-python), update your client to 0.29.0.

The following changes were made to the Enterprise API types:

- [Route](https://github.com/pomerium/enterprise-client/blob/main/API.md#route)
  - Removed `envoy_opts`
  - Added `load_balancing_policy`
  - Added `health_checks`
  - Replaced `RedirectAction`

Our full [Pomerium Enterprise gRPC API Reference](https://github.com/pomerium/enterprise-client/blob/main/API.md) is available for review as well.

#### Removed Audit Logging

> Audit logging is no longer supported. This rarely used feature is unrelated to authorization or access logging functionality that is still supported. If you rely on Pomerium's audit logging capability, reach out to us at [support@pomerium.com](mailto:support@pomerium.com) before upgrading to 0.29.0 or later.

Reviewer: And we may want to link to the [Authorize Log Fields](https://www.pomerium.com/docs/reference/authorize-log-fields) reference page as a potential replacement for the removed feature.

> Audit logging is no longer supported. This rarely used feature is unrelated to authorization or access logging functionality that is still supported. If you rely on Pomerium's audit logging capability, reach out to us at [support@pomerium.com](mailto:support@pomerium.com) before upgrading to 0.29.0 or later.

Author: Yes. I noticed the same thing. I'm pushing up an adjustment, but I'm open to suggestions on how to better communicate this.

Our full [Pomerium Enterprise gRPC API Reference](https://github.com/pomerium/enterprise-client/blob/main/API.md) is available for review as well.

#### Removed Decision Audit Logging

> Decision Audit logging with encryptedcontext preservation is no longer supported. This rarely used feature is unrelated to authorization or access logging functionality that is still supported. If you rely on Pomerium's decision audit logging capability, reach out to us at [support@pomerium.com](mailto:support@pomerium.com) before upgrading to 0.29.0 or later.

Reviewer: ```suggestion
Decision Audit logging with encrypted context preservation is no longer supported. This rarely used feature is unrelated to authorization or access logging functionality, which is still supported. If you rely on Pomerium's decision audit logging capability, reach out to us at [support@pomerium.com](mailto:support@pomerium.com) before upgrading to 0.29.0 or later.
```
